srcpkgs/wireplumber/files/wireplumber/run

```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+# this service is experimental and most setups should start pipewire and wireplumber as a user,
+# for further information, please refer to the handbook
+sv check pipewire > /dev/null || exit 1
```

Comment:
This probably doesn't do what you think it does. It certainly won't tell you if pipewire is running. I suspect order matters here, and pipewire needs to be running before wireplumber. If so, you'll need a more robust check.

Reply:
Not sure what robust check here is, because this global wireplumber is intended to run as long as global pipewire service is running. Do we need something like `sv check pipewire > /dev/null && (make sure pipewire has really been started before we proceed)`?

Reply:
`sv check pipewire` can't tell you if pipewire is running. It will only tell you if the service is linked in the service directory. For example, I believe `sv down pipewire` will still cause `sv check pipewire` to return success.

This works as most people expect for `sv check dbus` because `/etc/sv/dbus/check` exists and actually requires that the system bus be reachable before returning success. However, the pipewire service defines no `check` script so the check doesn't provide any real information.

If wireplumber *can* start before pipewire and will still do the right thing once the pipewire process is running, I would just drop the check altogether. If wireplumber dies when it starts before pipewire is running, dropping the check is OK too, because the service will just be retried until pipewire is up. However, if wireplumber starts but gets itself into an unrecoverable state (*e.g.*, starts its own pipewire process or simply hangs forever because it didn't find pipewire at startup) then something needs to be figured out to determine that pipewire is actually functional. This could be done here, or it could be done in an `/etc/sv/pipewire/check` script to make it available to all consumers.

Reply:
It actually exits if pipewire isn't running actually.

![a30-03-22-18h22m03s](https://user-images.githubusercontent.com/26714676/160838751-f28d30b9-78be-4f9c-9d14-de5b08f3d30e.jpg)